scripts/release.kts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Outputs the bash script that uploads packages to Bintray.
+ *
+ * This script assumes that all distrbution packages have been downloaded and unzipped in one directory.
+ * For example, all packages are in the "dist" directory:
+ *
+ * dist
+ *  ├── sentry-3.1.2-SNAPSHOT
+ *  ├── sentry-android-3.1.2-SNAPSHOT
+ *  ├── sentry-android-core-3.1.2-SNAPSHOT
+ *  ├── sentry-android-ndk-3.1.2-SNAPSHOT
+ *  ├── sentry-android-timber-3.1.2-SNAPSHOT
+ *  ├── sentry-log4j2-3.1.2-SNAPSHOT
+ *  ├── sentry-logback-3.1.2-SNAPSHOT
+ *  ├── sentry-servlet-3.1.2-SNAPSHOT
+ *  ├── sentry-spring-3.1.2-SNAPSHOT
+ *  └── sentry-spring-boot-starter-3.1.2-SNAPSHOT
+ *
+ * To execute the script two environment vairables that are used by Maven have to be present: BINTRAY_USERNAME, BINTRAY_API_KEY
+ *
+ * Example usage (assuming that the script is executed from the `<project-root>/scripts` directory and the distribution files are in `<project-root>/dist`):
+ * $ kotlinc -script release.kts -- -d ../dist -s  -repositoryUrl https://api.bintray.com/maven/sentry/sentry-java/sentry-java/ | sh
+ *
+ */
+import java.io.File
+
+/**
+ * Path to a directory with unzipped distribution packages.
+ */
+val path = argOrDefault("d", ".")
+
+/**
+ * Path to Maven settings.xml containing bintray username and api key.
+ */
+val settingsPath = argOrDefault("s", "./settings.xml")
+
+/**
+ * Bintray repository URL.
+ */
+val repositoryUrl = requiredArg("repositoryUrl")
+
+/**
+ * Maven server id in the settings.xml file.
+ */
+val repositoryId = argOrDefault("repositoryId", "bintray")
+
+/**
+ * If package should be published on bintray or just uploaded but not published.
+ */
+val publish = if (argOrDefault("publish", "false") == "true") 1 else 0
+
+File(path)
+    .listFiles { file -> file.isDirectory() }
+    .forEach { folder ->
+        val path = folder.path
+        val module = folder.name
+
+        val androidFile = folder.listFiles { file -> file.name.contains("release") && file.extension == "aar" }.firstOrNull()
+        val file = if (androidFile != null) {
+            androidFile.path
+        } else {
+            "$path/$module.jar"
+        }
+        val javadocFile = "$path/$module-javadoc.jar"
+        val sourcesFile = "$path/$module-sources.jar"
+        val pomFile = "$path/pom-default.xml"
+
+        val command = "mvn deploy:deploy-file -Dfile=$file -Dfiles=$javadocFile,$sourcesFile -Dclassifiers=sources,javadoc -Dtypes=jar,jar -DpomFile=$pomFile -DrepositoryId=$repositoryId -Durl=$repositoryUrl\\;publish\\=$publish --settings $settingsPath"
```

Comment:
this requires `mvn` installed which is fine, but afaik, bintray doesn't follow the mvn API conventions, unless they have something new, last time I've checked it, binary has its own APIs and CLIs to upload stuff.

Reply:
AFAIK it works with Maven `deploy-file` - check [SO question](https://stackoverflow.com/questions/42244931/how-can-i-deploy-artifacts-to-bintray-using-mvn-deploydeploy-file-command), I also tried it with my test project. We need to verify it with `sentry-java` to be sure.

Regarding Maven requirement, we can either expect Maven to be there or I can add Maven wrapper so that we will just call `./mvnw deploy:deploy-file ...` and proper Maven version will get installed. What do you think?

Reply:
if it works I'm happy with it.

The maven wrapper would be great, so nobody needs to care about the mvn version, etc, good idea.

Right now we are signing the files directly on bintray using bintray's key and autosign feature enabled here:
https://github.com/getsentry/sentry-java/blob/main/sentry/build.gradle.kts#L96

I believe we'd need to sign all the files locally (even poms) before publishing to bintray, which is gonna create a few more files (*.asc) and those should be uploaded too.

Not sure if we can pass this option via `deploy-file`.

Reply:
Can we give it a try? I think autosigning should work for Maven uploads too.

Reply:
yep we can try, but even on bintray's gradle plugin and bintray's CLI, this is opt-in, so I don't think it's gonna be enabled via deploy-file